### ISSUE TITLE
HLS errors

### ISSUE THREAD

**Issue by ctaity** [clappr/clappr#1809]
**Browser**: ALL

**OS**: ALL

**Clappr Version**: lastest

**Steps to reproduce**:
have a m3u8 link return 403

* I was expecting player call Clappr.Events.PLAYBACK_ERROR.

Hi, we have m3u8 with aes128 encription and we deny keys based on ips, browser etc. When we deny a key, clappr logs: 
  clappr.min.js:1 [warn][hlsjs: non-fatal error occurred, evt hlsError, data [object Object] ]
  clappr.min.js:1 [debug][hls] playback:levels:available (5) [{…}, {…}, {…}, {…}, {…}]

Take this error as non fatal error, but without key can play de the video. I think in this case the error is fatal.

**Comment by kslimani** [comment]
When i load a m3u8 returning a 403 forbidden i got a PLAYBACK_ERROR.

Using this code on [Clappr demo page](http://clappr.io/demo/) :

```javascript
var playerElement = document.getElementById("player-wrapper");

var player = new Clappr.Player({
  source: 'http://rbetis-hls.flumotion.com/rbetis/live/.m3u8',
  height: 360,
  width: 640,
  events: {
    onReady: function() {
      this.core.activePlayback.listenTo(
        this.core.activePlayback,
        Clappr.Events.PLAYBACK_ERROR,
        function(e) {
          console.log('playback_error', e);
        }
      );
    },
  },
});

player.attachTo(playerElement);
```

**Comment by ctaity** [comment]
Thanks for the reponse, but the error is on the aes 128 key request, we deny the key based on the ip for geolocation. 
 
On hls.js on the function   _onHLSJSError(evt, data), exits on: 
```
} else {
      error.level = PlayerError.Levels.WARN
      this.createError(error)
      Log.warn('hlsjs: non-fatal error occurred', { evt, data })
    }
```

I think, almost can trigger  playback error event with non-fatal to trap or something like thath. 
The error from HLS.JS lib is 

```javascript
  if (data.type == "networkError" && data.details == "keyLoadError") 
```

**Comment by kslimani** [comment]
Oh, i see.

I agree, denying the key in encrypted HLS source scenario could also trigger a playback fatal error. (it is technically near like returning 403 on manifest).

If you can share an url which reproduce the issue i can propose a fix. _(privately via email)_

**Comment by ctaity** [comment]
Yes, you can use: https://chunks01.tvpublica.com.ar/streams/BJQRNMF6b.m3u8 the stream is restricted only Argentina. If you outside argentina, this stream deny key.

**Comment by ctaity** [comment]
Yes, you can use: https://chunks01.tvpublica.com.ar/streams/BJQRNMF6b.m3u8 the stream is restricted only Argentina. If you outside argentina, this stream deny key.

**Comment by ctaity** [comment]
Yes but we use deny key, because we use a cdn with the chunks and the m3u8, and proccess key requests are few than proccess key requets, and you cant decrypt the stream withouth the key, without the m3u8 yo can recreate the m3u8. This is a requeriment of FIFA. Thanks

**Comment by towerz** [comment]
@ctaity @kslimani this behavior is the default for hls.js. You can see the same behavior pasting the test URL on https://hls-js.netlify.com/demo/.

I'm not sure if we should open this issue directly for hls.js or if we make a fix directly on Clappr. Either way, we could force the maximum number of retries (`hlsRecoverAttempts`) for network errors on the HLS playback.

**Comment by ctaity** [comment]
i think, some network errors  from hlsjs dont produce PLAYBACK ERRORS events on clappr, because this, need to lister errors directly from hlsjs, i think if you can trigger all network errors, on the else part of code i posted, is more simple control some features. 
Now on play event we listen ._hls object directly. But is a proposal only. 
Thanks.

**Comment by kslimani** [comment]
I think this is the default behaviour (ie: non-fatal) in hls.js to give to developer the choice to implements the mechanism which satisfy his need.

In the case of Clappr, we could imagine a new playback option, let's temporary call it `hlsTriggerErrorIfKeyDenied` which defaults to `false.` to preverse backward compatibility.

If set to `true` then it triggers a `PLAYBACK_ERROR` event if response code is 403. (which mean also it keep the current behaviour if, for example, key request return a 404 http error code).

What do you think @ctaity  @towerz ?

**Comment by ctaity** [comment]
For me is amazing!

**Comment by towerz** [comment]
@kslimani I think it's a good idea. Keeping backward compatibility is always a good thing.

I'd just suggest scoping the option inside the `playback` (or `playback.hls`) key instead of creating a new flat, prefixed option. This way we could have a clearer way to identify that the option only refers to the HLS playback behavior. About the name, perhaps we should trigger errors on all denied resource requests, so I'd suggest naming it `triggerFatalErrorOnResourceDenied`

Something like this:
```javascript
new Clappr.Player({
  // ...
  playback: {
    // this
    triggerFatalErrorOnResourceDenied: true,
    // or
    hls: {
      triggerFatalErrorOnResourceDenied: true,
    }
  }
})
```

**Comment by ctaity** [comment]
Yes is fine too.

**Comment by towerz** [comment]
@ctaity I just released `0.3.6` with the changes submitted by @kslimani. Please check again whether this solves your issue.

**Comment by ctaity** [comment]
Thanks. Monday i will try
